sys/cddl/dev/dtrace/mips/dtrace_isa.c

```diff
@@ -663,27 +676,38 @@ dtrace_next_uframe(register_t *pc, register_t *sp, register_t *ra)
 				switch (i.CCMType.cs) {
 				case OP_CINCOFFIMM:
 					if (i.CCMType.cb != 11 ||
-					    i.CCMType.rt != 11)
+					    i.CCMType.rt != 11 ||
+					    i.CCMType.offset > 0)
 						break;
+					/* cincoffsetimm   $c11,$c11,-X */
 					stksize = -((short)i.CCMType.offset);
 				}
 				break;
 			case OP_CSC:
-				//  csc     $c17,zero,X($c11)
-				if (i.CCMType.cb != 11 || i.CCMType.cs != 17 ||
-				    i.CCMType.rt != 0)
+				/* csc  $c17,zero,X($c11)
+				 * csc  $c24,zero,X($c11)
+				 * */
+				if (i.CCMType.cb != 11 ||
+				    (i.CCMType.cs != 17 && i.CCMType.cs != 24)
+				    || i.CCMType.rt != 0)
 					break;
-				// only first one
+				/* only the first */
 				if (mask & (1 << i.CCMType.cs))
 					break;
 				mask |= (1 << i.CCMType.cs);
 
 				void *base_addr_to_fetch =
 				    (void *)(vm_offset_t)(
 					*sp + (short)i.CCMType.offset * 16);
-
-				// Let's fetch only the address
-				*ra = dtrace_fuword64(base_addr_to_fetch + 8);
+				/* It is a capability. Let's skip the first 64
+				 * bytes with "+8" */
```

Comment:
`bits`.
It might be nicer to add a `dtrace_fucap` and use `cheri_getaddress()` on that value.